### DISCUSSION TITLE
Just a count ITEM

### DISCUSSION THREAD

**Discussion by maansch** [forntoh/LcdMenu#109] [Ideas]
Hi Thomas,
First of all, you really made a really great work ! Congratulations !
And also thank you for your help and these very fast fixes. :1st_place_medal: 

Is it possible to create a Item that just count up / down.
I use the liberty in my program witch has a lot of temperature and timer count downs.
With this functionality it will be easy to change the value very quick, ever with a rotary encoder.

Example:
Target temperature
 is 150°C  -> ENTER -> up or left  change incremental  to 163°C -> ENTER done.

Countdown timer 
Is 00:00:45 -> ENTER  -> up or left  change incremental  to 00:00:53 -> ENTER done.

Just two clicks.
Start value should be hand over from the program code.

From my view it will make a big step for end user usability comfort.
But I don’t know if it’s possible and how much work it is. 
br
Marc-Andre

**Comment by forntoh** [comment]
See #63, it is planned.

**Reply by maansch** [reply]
Oh, read the entire board but that i missed.

**Reply by forntoh** [reply]
Soon 😉

https://lcdmenu.forntoh.dev/examples/int-and-float-values
https://lcdmenu.forntoh.dev/reference/itemprogress
https://lcdmenu.forntoh.dev/useful-functions#mapprogress-uint16_t-version

#114

**Reply by maansch** [reply]
Hi
played around with the ItemProgress using the example code.
I noticed the following points:

intMapping:
initial value: LCD: 100m  -> right -> right 
LCD: 102m -> back
LCD: 102m output: 20
Enter -> left -> left
LCD: 100m output: 0
Enter -> right -> right -> right -> right
LCD: 104m -> back
LCD: 104m output: 40

LCD should be 110m, 120m, .. 140m ,
Output 140, right ?


floatMapping:
initial value: LCD: -1.00A -> right -> right
LCD: -0.98A   -> back
LCD: -0.98A output: 10
Enter -> right -> right ->
LCD: -0.96A   -> back
LCD: -0.96A output: 20

LCD should be -0.95A, -0.90A, .. -0.80A 
Output -0.80A, right ?
br
Marc-Andre

**Reply by forntoh** [reply]
Please read the documentation to see how it behaves, the stepLength determines how much each increment should go.

You should also map the value in the callback function, I didn't map it. It prints just the progress (unmapped)

**Reply by forntoh** [reply]
Creating two item types just to manage float and int values will use up memory because you will need to store the max value, min value current value, and step size as float AND as int for each item you create. Also, you will need a lot more code to handle both item types separately. The way I implemented it saves a lot of memory which you can use for your code, all you need to do is map the progress with the mapping functions provided.

**Reply by forntoh** [reply]
Check the documentation, you can also submit change requests if you feel the docs can be better.

**Comment by forntoh** [comment]
https://github.com/forntoh/LcdMenu/discussions/47#discussioncomment-6348694
